FlowCrypt/Controllers/Inbox/InboxViewController+Factory.swift

```diff
@@ -11,10 +11,9 @@ import UIKit
 
 class InboxViewControllerFactory {
     @MainActor
-    static func make(appContext: AppContext, with viewModel: InboxViewModel) -> InboxViewController {
+    static func make(appContext: AppContext, user: User, viewModel: InboxViewModel) -> InboxViewController {
         guard
-            let currentAuthType = appContext.dataService.currentAuthType,
-            let user = appContext.dataService.currentUser
+            let currentAuthType = appContext.dataService.currentAuthType
```

Comment:
This is another thing that should be dependency-injected. (separate issue, I suppose)

Also, maybe we should include `user` in the `AppContext`? That should minimize the changes, and clean things up.

Reply:
Agree - it seems we pass `user` param along with `appContext` almost each time, it'll be simpler and cleaner to have `user` property included in `AppContext`.

Reply:
And then, for situations that don't need `AppContext`, we'll pass the user alone

Reply:
Although - if we do this, I suppose the user in AppContext would have to be optional, so we are back where we started. Then each controller would have to still double check the user is there.

Could we then have two separate types - `AppContext` and `UserContext` ? The first one would stay as is, and `UserContext` would be required in all VCs that need it?

Reply:
I replaced several `user` with `UserContext` but it does not look better for me. Also it is hard to use two contexts because we don't have strict flow now like "here we 100% have session, user and auth type", and here not.

Reply:
See https://github.com/FlowCrypt/flowcrypt-ios/pull/1331#pullrequestreview-865505229

> don't have strict flow now like "here we 100% have session, user and auth type"

That will be the next step, another PR, another issue - all of that has to be refactored to have exactly that :+1: